compose/foundation/foundation/src/uikitMain/kotlin/androidx/compose/foundation/text/ContextMenu.uikit.kt

```diff
@@ -0,0 +1,257 @@
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.layout.positionInWindow
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.uikit.utils.CMPEditMenuView
+import androidx.compose.ui.unit.Density
+import kotlin.coroutines.resume
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.CancellableContinuation
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+
+/**
+ * Context menu area for [BasicTextField] (with [TextFieldValue] argument).
+ */
+@Composable
+internal actual fun ContextMenuArea(
+    manager: TextFieldSelectionManager,
+    content: @Composable () -> Unit
+) {
+    if (ComposeFoundationFlags.isNewContextMenuEnabled) {
+        // The first time the menu is called up, the menu item provider contains a non-final set of
+        // menu items, which causes the context menu callout to blink.
+        // Adding a small delay resolves this issue.
+        ProvideDefaultPlatformTextContextMenuProviders(
+            menuDelay = 100.milliseconds,
+            modifier = manager.contextMenuAreaModifier,
+            content = content
+        )
+    } else {
+        content()
+    }
+}
+
+/**
+ * Context menu area for [BasicTextField] (with [TextFieldState] argument).
+ */
+@Composable
+internal actual fun ContextMenuArea(
+    selectionState: TextFieldSelectionState,
+    enabled: Boolean,
+    content: @Composable () -> Unit
+) {
+    if (ComposeFoundationFlags.isNewContextMenuEnabled) {
+        val modifier = if (enabled) {
+            Modifier.textContextMenuGestures(
+                onPreShowContextMenu = { selectionState.updateClipboardEntry() }
+            )
+        } else {
+            Modifier
+        }
+        ProvideDefaultPlatformTextContextMenuProviders(
+            modifier = modifier,
+            content = content
+        )
+    } else {
+        CommonContextMenuArea(selectionState, enabled, content)
+    }
+}
+
+/**
+ * Context menu area for [SelectionContainer].
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+internal actual fun ContextMenuArea(
+    manager: SelectionManager,
+    content: @Composable () -> Unit
+) {
+    if (ComposeFoundationFlags.isNewContextMenuEnabled) {
+        ProvideDefaultPlatformTextContextMenuProviders(
+            modifier = manager.contextMenuAreaModifier,
+            content = content
+        )
+    } else {
+        CommonContextMenuArea(manager, content)
+    }
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun ProvideDefaultPlatformTextContextMenuProviders(
```

Comment:
We're creating a bit of a mess here.
There's already an `expect ProvideDefaultPlatformTextContextMenuProviders` function without the `menuDelay` argument, but we're not calling it here. In fact, it's neither actualized (except for an empty implementation) nor used in the uikit source set. So `CommonContextMenuArea` can't be used in this source-set.
@terrakok This looks messy. How should we clean it up?